src/software/ai/navigator/navigator.cpp

```diff
@@ -101,66 +84,54 @@ void Navigator::visit(const StopIntent &stop_intent)
 }
 
 // helpers
-void Navigator::movePointNavigation(const MoveIntent &move_intent,
-                                    std::vector<Point> &path_points)
+void Navigator::moveNavigation(const MoveIntent &move_intent, const Path &path)
 {
-    planned_paths.emplace_back(path_points);
-
-    switch (path_points.size())
+    if (path)
     {
-        case 0:
-        {
-            LOG(WARNING) << "Path planner could not find a path";
-            auto stop = std::make_unique<StopPrimitive>(move_intent.getRobotId(), false);
-            current_primitive = std::move(stop);
-            break;
-        }
-        case 1:
+        std::vector<Point> path_points = path->getKnots();
+        planned_paths.emplace_back(path_points);
+        if (path_points.size() == 1)
         {
             throw std::runtime_error(
                 "Path only contains one point, which is invalid, since it's ambiguous if it's the start or dest or some other point");
-            break;
         }
-        case 2:
+        if (path_points.size() == 2)
         {
-            current_destination = (path_points)[1];
+            current_destination = path_points[1];
             auto move           = std::make_unique<MovePrimitive>(
                 move_intent.getRobotId(), current_destination,
                 move_intent.getFinalAngle(),
                 move_intent.getFinalSpeed() *
-                    getCloseToEnemyObstacleFactor((path_points)[1]),
+                    getCloseToEnemyObstacleFactor(path_points[1]),
```

Comment:
I would like to see 'current_destination' used instead of the '1' here for legibility.

Reply:
I'll address this with nav cleanup